scripts/schemas/board-schema.yml

```diff
@@ -0,0 +1,73 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2023, Nordic Semiconductor ASA
+
+## A pykwalify schema for basic validation of the structure of a
+## board metadata YAML file.
+##
+# The board.yml file is a simple list of key value pairs containing board
+# information like: name, vendor, socs, variants.
+schema;variant-schema:
+  required: false
+  type: seq
+  sequence:
+    - type: map
+      mapping:
+        name:
+          required: true
+          type: str
+        cpucluster:
+          required: false
+          type: str
+        variants:
+          required: false
+          include: variant-schema
+
+type: map
+mapping:
+  board:
+    required: true
+    type: map
+    mapping:
+      name:
+        required: true
+        type: str
+        desc: Name of the board
+      vendor:
+        required: false
+        type: str
+        desc: SoC family of the SoC on the board.
+      revision:
+        required: false
+        type: map
+        mapping:
+          format:
+            required: true
+            type: str
+            enum:
+              ["major.minor.patch", "letter", "number", "fuzzy", "custom"]
+          default:
+            required: true
+            type: str
+          revisions:
+            required: true
+            type: seq
+            sequence:
+              - type: map
+                mapping:
+                  name:
+                    required: true
+                    type: str
+      socs:
+        required: false
+        type: seq
+        sequence:
+        - type: map
+          mapping:
+            name:
+              required: true
+              type: str
+            variants:
+              include: variant-schema
+      variants:
+        include: variant-schema
```

Comment:
Issue with x86/intel_adl board is that it has `crb` and `rvp` variants, there is no default, and it does not seem possible to use this without allowing a default. Also @nashif question for you: why is there another board in there which is not an intel board but is an "up squared pro 7000"?

Reply:
@nordicjm The Up2 Pro 7000 is almost identical to the two other Intel Alder Lake boards (CRB & RVP), so there'd have been lots of duplication if it had its own board directory. I guess you can think of `boards/x86/intel_adl` as "boards based on Intel's Alder Lake platform".

Reply:
Ignoring the duplicating, would you be okay with having it split into 3 board directories since these are actually 3 different boards, or at least having completely different names, e.g. `intel_adl_crb`, `intel_adl_rvp`, `up_squard..whatever it is...` rather than variants e.g. `intel_adl/crb`, `intel_adl/rvp`, `intel_adl/up_sq...`?

Reply:
I think the board schema should allow for multiple boards in same folder.

Was initially left out until I was fully sure it was a desired use-case.

Reply:
multiple boards in a single folder / board.yml is now supported, although the general guideline is that if such multiple boards in a single folder exists today, then those should probably be made into a single board with variants.